### PR TITLE
Move CI build cache instructions to Deployment docs

### PR DESCRIPTION
**Pull request by htunnicliff**
This PR moves the instructions for configuring CI providers to preserve Next.js's build cache out of the _No Cache Detected_ error message and into ~the _Deployment_~ a new page within the _Advanced Features_ docs. 

This change is beneficial because it makes the CI configuration examples visible in the context of other helpful documentation for deploying Next.js.

:point_right: **Prior to this change, these CI configuration examples didn't show up in the Next.js docs.**

The relocated instructions are now linked on the _No Cache Detected_ error message page.

## Documentation / Examples

- [x] Make sure the linting passes by running `yarn lint`

### CODE REVIEW

**Review by styfle — DISMISSED** [review]
Thanks!

**Review by leerob — CHANGES_REQUESTED** [review]
Agreed that we want to surface these steps outside of the errors page (good catch!) but we'll want to either:

1. Move this into its own page (it's long)
2. Condense and link to instructions from providers (I worry these are already out of date)

We're working to restructure this page of the docs as well, so let me think about this a bit more while we figure that out. Appreciate the contribution.

**Comment by htunnicliff** [comment]
@leerob that makes a lot of sense. 

I wonder if it would be helpful to add a provider-agnostic strategy for invalidating `.next/cache`? If that strategy was defined here, it could make these docs more evergreen since users could refer to provider-specific docs with the intended invalidation strategy in mind. 

E.g. "invalidate the Next.js cache if dependencies change or source files change" etc.

**Comment by htunnicliff** [comment]
Hey @leerob! Do you think it would be alright in the meantime to have this update go through as is? Reason being that folks can still benefit from this information being in the docs, even if it is eventually reorganized. 

Right now, this info doesn't exist in the docs anywhere, so just adding it in the first place would be really helpful.

**Comment by leerob** [comment]
- Let's make it a separate page "Advanced Features -> CI Build Caching"
    - There is a [manifest file](https://github.com/vercel/next.js/blob/canary/docs/manifest.json) that outlines the structure
- Update the link in the errors page to use this new page

**Comment by htunnicliff** [comment]
@leerob I was able to incorporate your requested changes – feel free to take a look and merge whenever you are ready!

**Comment by htunnicliff** [comment]
@leerob / @timneutkens – I think this PR is ready to go! 🚀 

Please let me know if there are any other changes I should make!

**Comment by htunnicliff** [comment]
Hi folks! Would you like me to close this PR? It has been several months and I haven't heard from anyone, so I'll plan on closing this PR a week from today if there is no word.

**Review by leerob — DISMISSED** [review]
Thank you! Sorry for the delay.

File: docs/advanced-features/ci-build-caching.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+description: Learn how to configure CI to cache Next.js builds
+---
+
+# Continuous Integration (CI) Build Caching
+
+To improve build performance, Next.js saves a cache to `.next/cache` that is shared between builds.
+
+To take advantage of this cache in Continuous Integration (CI) environments, your CI workflow will need to be configured to correctly persist the cache between builds.
+
+> If your CI is not configured to persist `.next/cache` between builds, you may see a [No Cache Detected](https://github.com/vercel/next.js/tree/canary/errors/no-cache.md) error.
```
Comment (styfle) [inline docs/advanced-features/ci-build-caching.md:11]:
```suggestion
> If your CI is not configured to persist `.next/cache` between builds, you may see a [No Cache Detected](https://nextjs.org/docs/messages/no-cache) error.
```

**Review by styfle — APPROVED** [review]
Thanks!

**Review by leerob — APPROVED** [review]

**Pull request merged**